test/odbc/test_pg_datatype_real.cpp

```diff
@@ -0,0 +1,920 @@
+#include <gtest/gtest.h>
+#include <sqlext.h>
+#include "odbc_handler.h"
+#include "query_generator.h"
+#include <cmath>
+#include <iostream>
+using std::pair;
+
+const string TABLE_NAME = "master_dbo.real_table_odbc_test";
+const string COL1_NAME = "pk";
+const string COL2_NAME = "data";
+const string DATATYPE_NAME = "sys.real";
+const string VIEW_NAME = "master_dbo.real_view_odbc_test";
+vector<pair<string, string>> TABLE_COLUMNS = {
+  {COL1_NAME, DATATYPE_NAME + " PRIMARY KEY"},
+  {COL2_NAME, DATATYPE_NAME}
+};
+const int DATA_COLUMN = 2;
+
+class PSQL_DataTypes_Real : public testing::Test{
+  void SetUp() override {
+    OdbcHandler test_setup;
+    test_setup.ConnectAndExecQuery(DropObjectStatement("TABLE", TABLE_NAME));
+  }
+
+  void TearDown() override {
+    OdbcHandler test_teardown;
+    test_teardown.ConnectAndExecQuery(DropObjectStatement("VIEW", VIEW_NAME));
+    test_teardown.CloseStmt();
+    test_teardown.ExecQuery(DropObjectStatement("TABLE", TABLE_NAME));
+  }
+};
+
+// Helper to convert string to float
+float StringToFloat(const string &value) {
+  return stof(value);
+}
+
+TEST_F(PSQL_DataTypes_Real, Table_Creation) {
+  // TODO - Expected needs to be fixed.
+  const int LENGTH_EXPECTED = 4;
+  const int PRECISION_EXPECTED = 0;      // Double check, Expect 6?
+  const int SCALE_EXPECTED = 0;
+  const string NAME_EXPECTED = "float4"; // Double check
+  
+  const int BUFFER_SIZE = 256;
+  char name[BUFFER_SIZE];
+  SQLLEN length;
+  SQLLEN precision;
+  SQLLEN scale;
+
+  RETCODE rcode;
+  OdbcHandler odbcHandler;
+
+  // Create a table with columns defined with the specific datatype being tested. 
+  odbcHandler.ConnectAndExecQuery(CreateTableStatement(TABLE_NAME, TABLE_COLUMNS));
+  odbcHandler.CloseStmt();
+
+  // Select * From Table to ensure that it exists
+  odbcHandler.ExecQuery(SelectStatement(TABLE_NAME, {"*"}, vector<string> {COL1_NAME}));
+
+  // Make sure column attributes are correct
+  rcode = SQLColAttribute(odbcHandler.GetStatementHandle(),
```

Comment:
sql_desc_length here looks like return the bytes, try also test the actual char size of the number since the main 3 attributes for numbers are precision scale and actual size.